From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: uazo <uazo@users.noreply.github.com>
Date: Wed, 19 Jul 2023 09:32:36 +0000
Subject: Compress libchrome to free up some space

upstream removed ModernLinker support for developer build (apk releases).
see https://bugs.chromium.org/p/chromium/issues/detail?id=1383210

License: GPL-2.0-or-later - https://spdx.org/licenses/GPL-2.0-or-later.html
---
 chrome/android/chrome_public_apk_tmpl.gni | 1 +
 chrome/android/java/AndroidManifest.xml   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/chrome/android/chrome_public_apk_tmpl.gni b/chrome/android/chrome_public_apk_tmpl.gni
index 2f545b1f2182813e36c8c329cedf17b2948af6e5..fc7975eda9f8dfeab7ac311c0f765e16b6f53f0e 100644
--- a/chrome/android/chrome_public_apk_tmpl.gni
+++ b/chrome/android/chrome_public_apk_tmpl.gni
@@ -584,6 +584,7 @@ template("chrome_common_apk_or_module_tmpl") {
         "//third_party/crashpad/crashpad/handler:crashpad_handler_trampoline",
       ]
       loadable_modules += [ "$root_out_dir/libcrashpad_handler_trampoline.so" ]
+      library_always_compress += [ "libchrome.so" ]
     }
 
     if (build_with_internal_optimization_guide) {
diff --git a/chrome/android/java/AndroidManifest.xml b/chrome/android/java/AndroidManifest.xml
index fa0eb501d7f4c03765f99a1be6573766f22c5e95..726ffff50aab4e2a3395c0b8e8396dec146f752e 100644
--- a/chrome/android/java/AndroidManifest.xml
+++ b/chrome/android/java/AndroidManifest.xml
@@ -200,6 +200,7 @@ by a child template that "extends" this file.
         android:roundIcon="@drawable/ic_launcher_round"
         android:label="{% block application_label %}@string/app_name{% endblock %}"
         android:memtagMode="async"
+        android:extractNativeLibs="true"
         android:largeHeap="false"
         android:manageSpaceActivity="@string/manage_space_activity"
         android:supportsRtl="true"
